llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2777,6 +2777,12 @@ Instruction *InstCombinerImpl::visitGEPOfGEP(GetElementPtrInst &GEP,
     Indices.append(GEP.idx_begin()+1, GEP.idx_end());
   }
 
+  // Don't create GEPs with more than one variable index.
+  unsigned NumVarIndices =
+      count_if(Indices, [](Value *Idx) { return !isa<Constant>(Idx); });
+  if (NumVarIndices > 1)
```

Comment:
Is there any reason for not putting the logic into `shouldMergeGEPs`? Do you mean the call to `simplifyAddInst` above may produce a constant?

Reply:
> Do you mean the call to `simplifyAddInst` above may produce a constant?

Yes, this is the motivation for the separate check.